gnovm/pkg/gnolang/op_binary.go

```diff
@@ -344,6 +343,9 @@ func isEql(store Store, lv, rv *TypedValue) bool {
 	} else if rvu {
 		return false
 	}
+	if err := checkSame(lv.T, rv.T, ""); err != nil {
```

Comment:
there's an issue with this case:

```go
package main

func main() {
	var l interface{}
	l = int64(0)

	switch val := l.(type) {
	case int64, int:
		println(val)
		if val == 0 {
			println("l is zero")
		} else {
			println("NOT zero")
		}
	}
}
```


when we have a clause with multi cases, the type is val is set as the type of the switchStmt, in this case, `interface{}`,  when preprocess this `if val == 0 `, the type of the RHS is given it's default type `int`. this will lead to a `not same type` in runtime when checking `checkSame`,  and a incorrect result of the comparison.

one solution is to not convert the untyped const to its default type when it's an comparison expression and the other side is of type interface, and do the `convertUntypedTo` in runtime. but, is it correct?